core/src/main/java/com/venia/core/models/commerce/services/ClearFullDispatcherCacheBasedOnCategory.java

```diff
@@ -0,0 +1,248 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClearFullDispatcherCacheBasedOnCategory.class);
+
+    // Constants for navigation structure
+    private static final String HEADER_FRAGMENT_PATH = "/content/experience-fragments/venia/us/en/site/header/master";
+    private static final String NAVIGATION_NODE_PATH = "jcr:content/root/navigation";
+    private static final String STRUCTURE_DEPTH_PROPERTY = "structureDepth";
+    private static final String LEVEL_KEY = "level";
+
+    @Override
+    public String getPattern() {
+        return null;
+    }
+
+    @Override
+    public String getInvalidationRequestType() {
+        return "categoryUids";
+    }
+
+    @Override
+    public List<String> getPathsToInvalidate(DispatcherCacheInvalidationContext context) {
+        if (context == null) {
+            LOGGER.warn("Context is null when getting paths to invalidate");
+            return Collections.emptyList();
+        }
+
+        // Get navigation structure depth from configuration
+        Integer navigationStructureDepth = getNavigationStructureDepth(context.getResourceResolver());
+        if (navigationStructureDepth == null) {
+            return Collections.emptyList();
+        }
+
+        // Extract and validate category UIDs
+        String[] categoryUids = extractCategoryUidsFromContext(context);
+        if (!isValidCategoryUids(categoryUids)) {
+            return Collections.emptyList();
+        }
+
+        // Fetch category data and determine if cache invalidation is needed
+        List<Map<String, Object>> categories = fetchCategoryData(context, categoryUids);
+        if (categories.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        if (shouldInvalidateFullCache(categories, navigationStructureDepth)) {
+            return Collections.singletonList(context.getStorePath());
+        }
+
+        return Collections.emptyList();
+    }
+
+    /**
+     * Extracts category UIDs from the context's attribute data.
+     *
+     * @param context The cache invalidation context
+     * @return Array of category UIDs
+     */
+    private String[] extractCategoryUidsFromContext(DispatcherCacheInvalidationContext context) {
+        if (context == null) {
+            LOGGER.warn("Context is null when extracting category UIDs");
+            return new String[0];
+        }
+
+        List<String> attributeData = context.getAttributeData();
+        return attributeData != null ? attributeData.toArray(new String[0]) : new String[0];
+    }
+
+    /**
+     * Fetches category information using GraphQL query.
+     *
+     * @param context The cache invalidation context
+     * @param categoryUids Array of category UIDs to query
+     * @return List of category data maps
+     */
+    private List<Map<String, Object>> fetchCategoryData(DispatcherCacheInvalidationContext context, String[] categoryUids) {
+        if (context == null || categoryUids == null) {
```

Comment:
Please remove all the checks which do not make sense. Don't make your example unnecessary complex and frightening for the customer.